mvn test reproduction lines do not work

When a test fails we print out the reproduction line, with the recent refactoring of the maven build logic, the line no longer works.

For example:

```
λ mvn verify -Pdev -Dskip.unit.tests -Dtests.seed=B8E9EE4B240CA1F3 -Dtests.class=org.elasticsearch.index.mapper.size.SizeMappingIT -Dtests.method="testBasic" -Des.logger.level=ERROR -Dtests.assertion.disabled=false -Dtests.security.manager=true -Dtests.heap.size=512m -Dtests.locale=th_TH_TH_#u-nu-thai -Dtests.timezone=Asia/Istanbul

[INFO] Executed tasks
[INFO] 
[INFO] --- maven-failsafe-plugin:2.18.1:verify (verify) @ elasticsearch ---
[INFO] Failsafe report directory: /home/hinmanm/src/elasticsearch/core/target/failsafe-reports
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] Elasticsearch Build Resources ...................... SUCCESS [  0.471 s]
[INFO] Elasticsearch Rest API Spec ........................ SUCCESS [  0.195 s]
[INFO] Elasticsearch Parent POM ........................... SUCCESS [  0.648 s]
[INFO] Elasticsearch Core ................................. FAILURE [  5.412 s]
[INFO] Elasticsearch Distribution ......................... SKIPPED
[INFO] Elasticsearch with all optional dependencies ....... SKIPPED
[INFO] Elasticsearch Shaded Distribution .................. SKIPPED
[INFO] Elasticsearch TAR Distribution ..................... SKIPPED
[INFO] Elasticsearch ZIP Distribution ..................... SKIPPED
[INFO] Elasticsearch DEB Distribution ..................... SKIPPED
[INFO] Elasticsearch RPM Distribution ..................... SKIPPED
[INFO] Elasticsearch Plugin POM ........................... SKIPPED
[INFO] Elasticsearch Japanese (kuromoji) Analysis plugin .. SKIPPED
[INFO] Elasticsearch Smart Chinese Analysis plugin ........ SKIPPED
[INFO] Elasticsearch Stempel (Polish) Analysis plugin ..... SKIPPED
[INFO] Elasticsearch Phonetic Analysis plugin ............. SKIPPED
[INFO] Elasticsearch ICU Analysis plugin .................. SKIPPED
[INFO] Elasticsearch Google Compute Engine cloud plugin ... SKIPPED
[INFO] Elasticsearch Azure cloud plugin ................... SKIPPED
[INFO] Elasticsearch AWS cloud plugin ..................... SKIPPED
[INFO] Elasticsearch Delete By Query plugin ............... SKIPPED
[INFO] Elasticsearch Python language plugin ............... SKIPPED
[INFO] Elasticsearch JavaScript language plugin ........... SKIPPED
[INFO] Elasticsearch Mapper size plugin ................... SKIPPED
[INFO] Elasticsearch example JVM plugin ................... SKIPPED
[INFO] Elasticsearch Example site plugin .................. SKIPPED
[INFO] QA: Parent POM ..................................... SKIPPED
[INFO] QA: Smoke Test Plugins ............................. SKIPPED
[INFO] QA: Smoke Test Shaded Jar .......................... SKIPPED
[INFO] QA: Smoke Test Multi-Node IT ....................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 7.450 s
[INFO] Finished at: 2015-08-12T11:38:29-06:00
[INFO] Final Memory: 60M/956M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-failsafe-plugin:2.18.1:verify (verify) on project elasticsearch: No tests were executed!  (Set -DfailIfNoTests=false to ignore this error.) -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :elasticsearch
```

Specifically: `Failed to execute goal org.apache.maven.plugins:maven-failsafe-plugin:2.18.1:verify (verify) on project elasticsearch: No tests were executed!`

Setting `-DfailIfNoTests=false` in the line does not work either.

@s1monw No, it looks like it no longer supports running integration test classes:

`12:57:34 ⏚ [jason:~/src/elastic/elasticsearch] master ± mvn clean verify -Dtests.class=org.elasticsearch.search.query.SearchQueryIT`

`[INFO] Elasticsearch Core ................................. FAILURE [ 36.993 s]`

`[ERROR] Failed to execute goal com.carrotsearch.randomizedtesting:junit4-maven-plugin:2.1.16:junit4 (tests) on project elasticsearch: There were no executed tests: 0 suites, 0 tests -> [Help 1]`

This is failing in core.

--- a/core/src/test/java/org/elasticsearch/test/junit/listeners/ReproduceInfoPrinter.java
+++ b/core/src/test/java/org/elasticsearch/test/junit/listeners/ReproduceInfoPrinter.java
@@ -78,9 +78,13 @@ public class ReproduceInfoPrinter extends RunListener {
 
         final StringBuilder b = new StringBuilder();
         if (inVerifyPhase()) {
-            b.append("REPRODUCE WITH: mvn verify -Pdev -Dskip.unit.tests");
+            b.append("REPRODUCE WITH: mvn verify -Pdev -Dskip.unit.tests" );
         } else {
             b.append("REPRODUCE WITH: mvn test -Pdev");
+        }
+        String project = System.getProperty("tests.project");
+        if (project != null) {
+            b.append(" -pl " + project);
         }
         MavenMessageBuilder mavenMessageBuilder = new MavenMessageBuilder(b);
         mavenMessageBuilder.appendAllOpts(failure.getDescription());

--- a/pom.xml
+++ b/pom.xml
@@ -623,6 +623,7 @@
                             <tests.appendseed>${tests.appendseed}</tests.appendseed>
                             <tests.cluster>${tests.cluster}</tests.cluster>
                             <tests.iters>${tests.iters}</tests.iters>
+                            <tests.project>${project.groupId}:${project.artifactId}</tests.project>
                             <tests.maxfailures>${tests.maxfailures}</tests.maxfailures>
                             <tests.failfast>${tests.failfast}</tests.failfast>
                             <tests.class>${tests.class}</tests.class>

